pkg/drift_client/drift_client.py

```
@@ -111,6 +114,53 @@ def get_list(self, topics: List[str], timeframe: List[str]) -> Dict[str, List[st
 
         return data
 
+    def get_topic_data(self, topics: List[str], start, stop) -> Dict[str, List[str]]:
+        """Returns list of history data from initialised Device
+
+        Args:
+            topics: List of topic names, e.g. `["sensor-1", "sensor-2"]`
+            start: Begin of request timeframe,
+                Format: ISO string, datetime or float timestamp
+            stop: End of request timeframe,
+                Format: ISO string, datetime or float timestamp
+
+        Returns:
+            Dict with topics and item names available
+        :rtype: Dict[str, List[str]]
+
+        Examples:
+            >>> client = DriftClient("127.0.0.1", "PASSWORD")
+            >>> client.get_topic_data(["topic-1", "topic-2", "topic-3"],
+            >>>         "2022-02-03 10:00:00", "2022-02-03 10:00:10")
+            >>> # => {"topic-1": ['topic-1/1644750600291.dp',
+            >>> #                  'topic-1/1644750601291.dp', ...] ... }
+        """
+        try:
+            return self._get_topic_data(
+                topics,
+                start.strftime("%Y-%m-%d %H:%M:%S"),
+                stop.strftime("%Y-%m-%d %H:%M:%S"),
+            )
+        except (TypeError, AttributeError):
+            pass
+
+        try:
+            start_iso = datetime.fromisoformat(start).strftime("%Y-%m-%d %H:%M:%S")
+            stop_iso = datetime.fromisoformat(stop).strftime("%Y-%m-%d %H:%M:%S")
+            return self._get_topic_data(topics, start_iso, stop_iso)
+        except (TypeError, AttributeError):
+            pass
```

we don't suppress exceptions! You make a user blind.